standards/SPIFFE_Trust_Domain_and_Bundle.md

```diff
@@ -0,0 +1,290 @@
+
+The first and most obvious is that you are assuming trust in the public certificate authorities and in their fulfillment procedures. Publicly trusted CAs have the technical ability to sign certificates which will be broadly trusted and therefore could, in theory, sign a valid certificate for the host name used to serve your bundle endpoint. While this may be an unacceptable risk to some, most users are okay with it given the convenience it provides and the historical stability of the public CA infrastructure.
+
+Second, more importantly, recently introduced automated fulfillment protocols like [ACME][9] make room for attacks that were not previously possible. ACME allows publicly trusted certificates to be issued without manual intervention by performing a challenge which allows a machine to assert that it is authorized by answering an HTTP request made to the DNS name of the certificate being issued. This is significant because it means that anyone that can intercept and answer the challenge is capable of receiving a valid publicly-trusted certificate. While the risk of interception across the open internet is decidedly low, risk of interception by a neighboring machine is high. Concretely, without adequate Layer 2 security controls, SPIFFE bundle endpoints using Web PKI may be subverted by malicious software with access to the same Layer 2 network as the bundle endpoint. Operators should be careful in ensuring that Layer 2 access to SPIFFE bundle endpoints protected by Web PKI is restricted.
+
+### 6.4. Reusing Cryptographic Keys Across Trust Domains
+This specification discourages sharing cryptographic keys across trust domains since the practice degrades trust domain isolation and introduces additional security challenges. When a root key is shared across multiple trust domains, it becomes critically important that authentication and authorization implementations carefully check the trust domain component of an identity and that the trust domain component be easily and habitually expressed in authorization policies.
+
+Suppose that a naïve implementation imported (ie. fully trusted) a particular root key and that the authentication system were configured to authenticate the SPIFFE identity of any SVID which chained up to the trusted root key. If the naïve implementation is not configured to only trust a specific trust domain, then any identity issued in any trust domain could be authenticated (so long as the SVID chains up to the trusted root key). Successful authentication of an external trust domain is not a problem per se since the risk can be mitigated by a correctly implemented authorization system; however, this risk could alternatively be mitigated by simply not reusing root keys across trust domains.
```

Comment:
```suggestion
Suppose that a naïve implementation imported (ie. fully trusted) a particular root key and that the authentication system were configured to authenticate the SPIFFE identity of any SVID which chained up to the trusted root key. If the naïve implementation is not configured to only trust a specific trust domain, then any identity issued in any trust domain could be authenticated (so long as the SVID chains up to the trusted root key).
```

I think we should just strike the last sentence, and avoid planting the seed of the idea that having each trust domain have a unique key is an appropriate alternative to robustly implemented authentication.

The message is
 a) we recommend you don't share keys across trust domains
 b) _if you are_ going to share keys, you better make damned sure that authn and authz are robust and trust-domain aware (which you should be doing anyway)

Reply:
Thanks Spike. I think that @justinburke has already stated that he's in agreement with this change so I went ahead and applied it (Justin, please chime in if that's not the case)